komodo-relational/src/main/java/org/komodo/relational/workspace/WorkspaceManager.java

```diff
@@ -184,15 +260,16 @@ public KomodoType getTypeIdentifier(UnitOfWork uow) {
      *
      * @param repository remove instance with given repository
      */
-    public static void uncacheInstance(final Repository repository) {
+    public static void uncacheInstance(final Repository repository, final String owner) {
         if (repository == null)
             return;
 
-        instances.remove(repository.getId());
+        instances.remove(new CacheKey(repository.getId(), owner));
     }
 
-    private WorkspaceManager(UnitOfWork uow, Repository repository ) throws KException {
-        super( repository, RepositoryImpl.WORKSPACE_ROOT, 0 );
+    private WorkspaceManager(Repository repository, UnitOfWork uow ) throws KException {
```

Comment:
komodoWorkspacePath and the methods is calls on all require the transaction. Yes, the username is extracted at the 'end' but its still a chain to get there. Also, I really prefer not passing around a String representing a username as
a) its a String so anything can be stuck in it and looks vague
b) want to avoid passing around Repository.SYSTEM_USER since at some point I bet that object will get changed and improved.